archive/schema.go

```diff
@@ -107,25 +123,97 @@ func (c *CreateOptions) toMetadata() *Metadata {
 }
 
 type Archive struct {
-	Meta *Metadata
-	Root string
+	Root              string
+	DataSortDirection zbuf.Direction
```

Comment:
I think this is a good thing to do regardless, but is replacing `Archive.Meta` by  `Archive{DataSortDirection,LogSizeThreshold}`

Reply:
On an earlier version of this change, I had a bug where I used the wrong spans slice -  `Archive.Spans` (for filtered spans) vs `Archive.Meta.Spans` (for all the available spans). So I wanted to remove the metadata pointer from `Archive` to prevent tripping up on that again.